IdentityCore/tests/MSIDLastRequestTelemetryTests.m

```diff
@@ -260,4 +262,68 @@ - (void)testSaveToDisk_whenManySilentCalls_shouldOverwriteAndRestoreToSameObject
     XCTAssertEqualObjects([restoredTelemetryObject telemetryString], [telemetryObject telemetryString]);
 }
 
+- (void)testSerializationSizeLimit_whenStringTooLong_shouldBreakUpIntoMulitpleRequests
+{
+    MSIDLastRequestTelemetry *telemetryObject = [MSIDLastRequestTelemetry sharedInstance];
+    
+    [MSIDLastRequestTelemetry updateTelemetryStringSizeLimit:200];
+    
+    int errorNum = 10;
+    for (int i = 1; i <= errorNum; i++)
+    {
+        NSString *errorString = [NSString stringWithFormat:@"error%d", i];
+        [telemetryObject updateWithApiId:i errorString:errorString context:self.context];
+    }
+    
+    dispatch_queue_t queue = [telemetryObject valueForKey:@"synchronizationQueue"];
+    MSIDLastRequestTelemetry *restoredTelemetryObject = [[MSIDLastRequestTelemetry alloc] initTelemetryFromDiskWithQueue:queue];
+    
+    NSString *result = [restoredTelemetryObject telemetryString];
+    // Simulates a successful request to server, string up to limit size is sent to server
+    [restoredTelemetryObject updateWithApiId:0 errorString:nil context:self.context];
+    
+    NSUInteger afterFirstCutoff = restoredTelemetryObject.errorsInfo.count;
+    XCTAssertTrue(afterFirstCutoff > 0);
+    XCTAssertTrue(afterFirstCutoff < errorNum);
+    
+    // Check if least recently added errors are left over after telemetry is sent to server
+    for (int i = 0; i < afterFirstCutoff; i++)
+    {
+        XCTAssertEqualObjects([restoredTelemetryObject.errorsInfo objectAtIndex:i].error, [telemetryObject.errorsInfo objectAtIndex:i].error);
+    }
+    
+    
+    result = [restoredTelemetryObject telemetryString];
+    // Simulates a successful request to server, string up to limit size is sent to server
+    [restoredTelemetryObject updateWithApiId:1 errorString:nil context:self.context];
+    
+    NSUInteger afterSecondCutoff = restoredTelemetryObject.errorsInfo.count;
+    XCTAssertTrue(afterSecondCutoff > 0);
+    XCTAssertTrue(afterSecondCutoff < afterFirstCutoff);
+    
+    for (int i = 0; i < afterSecondCutoff; i++)
+    {
+        XCTAssertEqualObjects([restoredTelemetryObject.errorsInfo objectAtIndex:i].error, [telemetryObject.errorsInfo objectAtIndex:i].error);
+    }
+}
+
+- (void)testArchiveSizeLimit_whenObjectTooBig_shouldCutOffLeastRecentTelemetry
+{
+    MSIDLastRequestTelemetry *telemetryObject = [MSIDLastRequestTelemetry sharedInstance];
```

Comment:
Just to solidify my understanding; so in this particular scenario, the first n errorString would be cut, since they are least recent (added first = older)?

Reply:
Exactly. In future releases, it would be useful to fine tune this idea to be something like: cut older (added first) telemetry UNLESS the more recently added telemetry is the same error over and over, or UNLESS the error is something like NetworkUnavailable, which maybe we would try to squash multiple instances of into just one error in the telemetry object